### PR TITLE
Task04v2

### PR DESCRIPTION
**Pull request by bertei**
New branch and task04 version with some of your pull requests.

### CODE REVIEW

File: wordpress.yaml [hunk 1]
```diff
@@ -1,7 +1,104 @@
...
+        src: /etc/nginx/sites-available/wordpressCfg.txt
+        dest: /etc/nginx/sites-enabled/wordpressCfg.txt
+        state: link
+        
+    - name: Unlink Default nginx config      
+      file:
+        path: /etc/nginx/sites-enabled/default
+        state: absent
+        
+    - name: Create Wordpress Database
```
Comment (korney4eg) [inline wordpress.yaml]: have you seen https://docs.ansible.com/ansible/latest/collections/community/mysql/mysql_db_module.html ?

File: wordpress.yaml [hunk 2]
```diff
@@ -1,7 +1,104 @@
...
+        state: absent
+        
+    - name: Create Wordpress Database   
+      shell: 'mysql < wpDB.sql'
+      args:
+        chdir: /vagrant/
+        
+    - name: Download Wordpress        
+      get_url:
+        url: https://wordpress.org/latest.tar.gz
```
Comment (korney4eg) [inline wordpress.yaml]: still latest?

File: wordpress.yaml [hunk 3]
```diff
@@ -1,7 +1,104 @@
...
+        mode: 0755
+            
+    - name: Download WP-Cli file      
+      get_url:
+        url: https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar
+        dest: /usr/local/bin/wp
+        mode: 0755
+    
+    - name: Install WP-CLI     
+      shell: 'wp core install --url="192.168.50.2" --title="Wordpress" --admin_user=admin --admin_email=admin@admin123.com --admin_password="!2three456." --path=/var/www/wordpress --allow-root'
```
Comment (korney4eg) [inline wordpress.yaml]: can you create separate user, like `wordpress` or `wp` or reuse existing and not run commands with root privileges?

File: nginx.conf.j2 [hunk 4]
```diff
@@ -0,0 +1,44 @@
+server {
```
Comment (korney4eg) [inline nginx.conf.j2:1]: where is this template used?

File: wordpress.yaml (hunk 3 above)
Comment (korney4eg) [reply wordpress.yaml]: Most of this things are described with variables. Could you use them here?

File: wp-config.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
+<?php
+
+/** The name of the database for WordPress */
+define( 'DB_NAME', 'wpDB' );
```
Comment (korney4eg) [inline wp-config.php]: Most of this things are described with variables. Could you use them here?

File: wp-config.php.j2 [hunk 6]
```diff
@@ -0,0 +1,51 @@
+<?php
```
Comment (korney4eg) [inline wp-config.php.j2:1]: where is it used?

File: wordpress.yaml (hunk 1 above)
Comment (bertei) [reply wordpress.yaml]: I found easier to install mysql database from an .sql configuration file. I could try that too.

File: nginx.conf.j2 [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    }
+	
+	
+}
+
+server {
+    listen 80;
+    listen [::]:80;
+
+    server_name 192.168.50.2;
```
Comment (korney4eg) [inline nginx.conf.j2]: can you use variable here?

File: wordpress.yaml [hunk 8]
```diff
@@ -1,7 +1,137 @@
...
+    - /vagrant/variables.yaml
+    
   tasks:
-
-    - name: message to remove
-      command: echo 'Here you gonna put all tasks and roles that gonna be runing on the VM'
+    - name: Update & Upgrade the System
+      apt:
+        update_cache: yes
+        upgrade: yes
```
Comment (korney4eg) [inline wordpress.yaml]: please remove `upgrade` here

File: wordpress.yaml [hunk 9]
```diff
@@ -1,7 +1,137 @@
...
+            - php7.4-xml
+            - php7.4-gd
+            - php7.4-curl
+            
+    - name: Allow NGINX on Firewall   
+      ufw:
+        rule: allow
+        name: "Nginx Full"
+        
+    - name: Create SSL-Certificate
```
Comment (korney4eg) [inline wordpress.yaml:71]: it creates new ssl certificate every time that `vagrant deploy` is running. Could you make it run only one time?

File: wordpress.yaml [hunk 10]
```diff
@@ -1,7 +1,137 @@
...
+      shell: |
+            openssl req -x509 -newkey rsa:2048 -days 365 -nodes \
+                -subj "/C=AR/ST=BSAS/L=BSAS/O=WordPress/OU=WP/CN=192.168.50.2" \
```
Comment (korney4eg) [inline wordpress.yaml]: please remove hardcoded values and use variables instead

File: wordpress.yaml [hunk 11]
```diff
@@ -1,7 +1,137 @@
...
+        name: "{{ db_name }}"
+        state: present
+        login_user: root
+        login_password: "{{ mysql_password }}"
+    
+    - name: Create WordPress Database user
+      mysql_user:
+        name: "{{ db_user }}"
+        password: "{{ db_password }}"
+        priv: "*.*:ALL,GRANT"
```
Comment (korney4eg) [inline wordpress.yaml]: here you give all privileges to all databases. I think your user needs access only to your wordpress database.

File: wordpress.yaml [hunk 12]
```diff
@@ -1,7 +1,137 @@
...
+        state: present
+        login_user: root
+        login_password: "{{ mysql_password }}"
+        
+    - name: Restart MySQL
```
Comment (korney4eg) [inline wordpress.yaml]: you want to restart database every run?

File: wordpress.yaml [hunk 13]
```diff
@@ -1,7 +1,137 @@
...
+        mode: 0755
+            
+    - name: Download WP-Cli file      
+      get_url:
+        url: "{{ url_wpcli }}"
+        dest: /usr/local/bin/wp
+        mode: 0755
+    
+    - name: Install WP-CLI     
+      shell: 'wp core install --url="{{ wp_iphost }}" --title="{{ title }}" --admin_user="{{ wp_admin_user }}" --admin_email="{{ wp_admin_mail }}" --admin_password="{{ wp_admin_password }}" --path="/var/www/{{ wp_root_dir }}" --allow-root'
```
Comment (korney4eg) [inline wordpress.yaml]: please do not run `wp` as root

File: wordpress.yaml [hunk 14]
```diff
@@ -1,7 +1,137 @@
...
+      
+    - name: Install Theme
```
Comment (korney4eg) [inline wordpress.yaml]: please do not run wp as root

File: wordpress.yaml [hunk 15]
```diff
@@ -1,7 +1,137 @@
...
+        dest: /usr/local/bin/wp
+        mode: 0755
+    
+    - name: Install WP-CLI     
+      shell: 'wp core install --url="{{ wp_iphost }}" --title="{{ title }}" --admin_user="{{ wp_admin_user }}" --admin_email="{{ wp_admin_mail }}" --admin_password="{{ wp_admin_password }}" --path="/var/www/{{ wp_root_dir }}" --allow-root'
+      
+    - name: Install Theme     
+      shell: 'wp theme install twentynineteen --activate --path="/var/www/{{ wp_root_dir }}" --allow-root'
+    
+    - name: Restart MySQL
```
Comment (korney4eg) [inline wordpress.yaml]: you probably wanted to name it as `Restart Nginx`.

File: wordpress.yaml [hunk 16]
```diff
@@ -1,7 +1,137 @@
...
+      
+    - name: Install Theme     
+      shell: 'wp theme install twentynineteen --activate --path="/var/www/{{ wp_root_dir }}" --allow-root'
+    
+    - name: Restart MySQL  
+      service:
+        name: nginx
+        state: restarted
+    
+    - name: Restart Nginx
```
Comment (korney4eg) [inline wordpress.yaml]: duplicate?

File: wordpress.yaml [hunk 17]
```diff
@@ -1,7 +1,137 @@
...
+    - name: Install WP-CLI     
+      shell: 'wp core install --url="{{ wp_iphost }}" --title="{{ title }}" --admin_user="{{ wp_admin_user }}" --admin_email="{{ wp_admin_mail }}" --admin_password="{{ wp_admin_password }}" --path="/var/www/{{ wp_root_dir }}" --allow-root'
+      
+    - name: Install Theme     
+      shell: 'wp theme install twentynineteen --activate --path="/var/www/{{ wp_root_dir }}" --allow-root'
+    
+    - name: Restart MySQL  
+      service:
+        name: nginx
+        state: restarted
```
Comment (korney4eg) [inline wordpress.yaml]: do you want to restart nginx every run?

**Review by korney4eg — CHANGES_REQUESTED** [review]

File: wordpress.yaml (hunk 14 above)
Comment (bertei) [reply wordpress.yaml]: Do you mean to delete: '--allow-root' line?

File: nginx.conf.j2 (hunk 4 above)
Comment (bertei) [reply nginx.conf.j2:1]: On /etc/nginx. Its the main configuration file for nginx. Where you point the server domain, ports to use, certification, etc.

File: wp-config.php.j2 (hunk 6 above)
Comment (bertei) [reply wp-config.php.j2:1]: File to connect php with mysql database.

File: wordpress.yaml (hunk 14 above)
Comment (korney4eg) [reply wordpress.yaml]: yes and run it from separate user, other then root.

File: wordpress.yaml [hunk 18]
```diff
@@ -1,7 +1,149 @@
...
+            - php7.4-curl
+            
+    - name: Allow NGINX on Firewall   
+      ufw:
+        rule: allow
+        name: "Nginx Full"
+        
+    - name: Create SSL-Certificate
+      shell: |
+                if [ ! -f /etc/ssl/certs/nginx-selfsigned.crt ]; then
```
Comment (korney4eg) [inline wordpress.yaml]: ansible has great feature to run this if file not exists. Could you check it out?

File: wordpress.yaml [hunk 19]
```diff
@@ -1,7 +1,149 @@
...
+        name: "{{ db_name }}"
+        state: present
+        login_user: root
+        login_password: "{{ mysql_password }}"
+    
+    - name: Create WordPress Database user
+      mysql_user:
+        name: "{{ db_user }}"
+        password: "{{ db_password }}"
+        priv: "wpDB.*:ALL,GRANT"
```
Comment (korney4eg) [inline wordpress.yaml]: could you remove hardcoded database name here and use variable instead?

File: wordpress.yaml [hunk 20]
```diff
@@ -1,7 +1,149 @@
...
+        state: present
+        login_user: root
+        login_password: "{{ mysql_password }}"
+        
+    - name: Extract Wordpress file
```
Comment (korney4eg) [inline wordpress.yaml]: so you extract archive and overwrite wordpress files every run?

File: wordpress.yaml [hunk 21]
```diff
@@ -1,7 +1,149 @@
...
+        mode: 0755
+    
+    - name: Setup 777 permissions to wp-content, so upload works
+      file:
+        path: /var/www/{{ wp_root_dir }}
+        state: directory
+        recurse: yes
+        owner: www-data
+        group: www-data
+        mode: 0777
```
Comment (korney4eg) [inline wordpress.yaml]: what gives mode 777 here? I mean what is the reason to set it here?

File: wordpress.yaml [hunk 22]
```diff
@@ -1,7 +1,149 @@
...
+            
+    - name: Download WP-Cli file      
+      get_url:
+        url: "{{ url_wpcli }}"
+        dest: /usr/local/bin/wp
+        mode: 0755
+    
+    - name: Install WP-CLI     
+      shell: 'sudo -u ber -i -- wp core install --url="{{ wp_iphost }}" --title="{{ title }}" --admin_user="{{ wp_admin_user }}" --admin_email="{{ wp_admin_mail }}" --admin_password="{{ wp_admin_password }}" --path="/var/www/{{ wp_root_dir }}"'
```
Comment (korney4eg) [inline wordpress.yaml]: in ansible there is a proper option to run command as user. Also it's not a good habbit to run same command every time. I suppose you want to run it once when there is no wordpress running

File: wordpress.yaml [hunk 23]
```diff
@@ -1,7 +1,149 @@
...
+      
+    - name: Install Theme     
+      shell: 'sudo -u ber -i -- wp theme install twentynineteen --activate --path="/var/www/{{ wp_root_dir }}"'
```
Comment (korney4eg) [inline wordpress.yaml]: in ansible there is a proper option to run command as user. Also it's not a good habbit to run same command every time. I suppose you want to run it once when there is no wordpress running

File: wordpress.yaml (hunk 17 above)
Comment (korney4eg) [reply wordpress.yaml]: why do you restart mysql server every run? What the point of it?

File: wordpress.yaml [hunk 24]
```diff
@@ -1,7 +1,149 @@
...
+    
+    - name: Restart MySQL  
+      service:
+        name: mysql
+        state: restarted
+    
+    - name: Restart Nginx  
+      service:
```
Comment (korney4eg) [inline wordpress.yaml]: why do you restart nginx every time. I suppose you want to do it only when configuration files changed.

**Review by korney4eg — CHANGES_REQUESTED** [review]

File: wordpress.yaml (hunk 21 above)
Comment (bertei) [reply wordpress.yaml]: My bad

File: wordpress.yaml (hunk 14 above)
Comment (bertei) [reply wordpress.yaml]: done

File: wordpress.yaml (hunk 18 above)
Comment (bertei) [reply wordpress.yaml]: done

**Comment by bertei** [comment]
up

**Comment by bertei** [comment]
Finished.

**Pull request closed without merging**